### PR TITLE
Improve the method isValidGlobalName

### LINKED ISSUE #7532: isValidGlobalName should be updated and improve
**Describe the bug**
`MyClass` is apparently now used in class template, and therefore, it cannot be used as a global variable name.

**To Reproduce**
1. `'MyClass' isValidGlobalName` should returns `false`

### PR DESCRIPTION
**Pull request by bergel**
Fix #7532

### CODE REVIEW

**Comment by MarcusDenker** [comment]
failing tests, e.g:

FluidClassBuilderTest>>testCreateClassWithFullExpandedDefinitionKeepsTheMinimum

File: src/Collections-Strings/String.class.st [hunk 1]
```diff
@@ -1874,17 +1874,16 @@ String >> isString [
...
 { #category : #testing }
 String >> isValidGlobalName [
-
+	"Answer whether the string receiver can be used as a global name (e.g., name of a class, name of a trait)"
 	self ifEmpty: [ ^ false ].
 	
 	"reserverd default names"
-	self = 'NameOfSubclass' ifTrue: [ ^ false ].
-	self = 'TNameOfTrait' ifTrue: [ ^ false ].
+	(#('NameOfSubclass' 'MyClass' 'TNameOfTrait') includes: self) ifTrue: [ ^ false ].
```
Comment (dionisiydk) [inline src/Collections-Strings/String.class.st:1881]: I did not know about this logic. But it's very surprizing. Why we can't have classes with those names? What makes them so special?

Comment (dionisiydk) [reply src/Collections-Strings/String.class.st:1881]: It would be really nice to explain it in the method comment

Comment (astares) [reply src/Collections-Strings/String.class.st:1881]: @bergel also it should be "reserved" - not "reserverd"

**Comment by guillep** [comment]
@bergel there are broken tests and pending comments in the PR, and a couple of months of inactivity. I'll close it for now, please do not hesitate to reopen it.

**Comment by astares** [comment]
@guillep one can not reopen as outside contributor due to permissions.

**Comment by guillep** [comment]
I can reopen it myself if required, no problem :)

**Pull request closed without merging**